index.js

```diff
@@ -96,48 +85,8 @@ module.exports = terria
       updateApplicationOnMessageFromParentWindow(terria, window);
 
       // Show a modal disclaimer before user can do anything else.
-      if (defined(terria.configParameters.globalDisclaimer)) {
-        var globalDisclaimer = terria.configParameters.globalDisclaimer;
-        var hostname = window.location.hostname;
-        if (
-          globalDisclaimer.enableOnLocalhost ||
-          hostname.indexOf("localhost") === -1
-        ) {
-          var message = "";
-          // Sometimes we want to show a preamble if the user is viewing a site other than the official production instance.
-          // This can be expressed as a devHostRegex ("any site starting with staging.") or a negative prodHostRegex ("any site not ending in .gov.au")
-          if (
-            (defined(globalDisclaimer.devHostRegex) &&
-              hostname.match(globalDisclaimer.devHostRegex)) ||
-            (defined(globalDisclaimer.prodHostRegex) &&
-              !hostname.match(globalDisclaimer.prodHostRegex))
-          ) {
-            message += require("./lib/Views/DevelopmentDisclaimerPreamble.html");
-          }
-          message += require("./lib/Views/GlobalDisclaimer.html");
-
-          var options = {
-            title:
-              globalDisclaimer.title !== undefined
-                ? globalDisclaimer.title
-                : "Warning",
-            confirmText: globalDisclaimer.buttonTitle || "Ok",
-            denyText: globalDisclaimer.denyText || "Cancel",
-            denyAction: globalDisclaimer.afterDenyLocation
-              ? function () {
-                  window.location = globalDisclaimer.afterDenyLocation;
-                }
-              : undefined,
-            width: 600,
-            height: 550,
-            message: message,
-            horizontalPadding: 100
-          };
-          runInAction(() => {
-            viewState.disclaimerSettings = options;
-            viewState.disclaimerVisible = true;
-          });
-        }
+      if (terria.configParameters.globalDisclaimer !== undefined) {
```

Comment:
My impression for how TSification is done in TerriaJS is that `defined()` is replaced with `isDefined()`, so leaving the `defined()` should make it easier to convert this to TypeScript in the future.